src/main/ts/components/Chart.tsx

```diff
@@ -49,25 +53,53 @@ export default function Chart() {
     caycStartingDate,
     chartStartingDate,
     chartEndDate,
+    setSelectedProjects,
   } = useData();
 
-  if (isLoading) {
-    return <Spinner />;
-  }
+  const handleLoadProjects = async (q: string, resolve: (options: ProjectOption[]) => void) => {
+    const components = await getProjects({ q });
+    if (!components || components.length === 0) {
+      resolve([]);
+      return [];
+    }
+    const options = [
+      defaultProject,
+      ...components.map(({ key, name }) => {
+        return {
+          value: key,
+          label: name,
+        } as ProjectOption;
+      }),
+    ];
+    resolve(options);
+    return options;
+  };
```

Comment:
The resolve thing makes it really hard to understand what's happening. This is much simpler I think.
```suggestion
  const handleLoadProjects = (q: string) =>
    getProjects({ q }).then((components) => [
      defaultProject,
      ...components.map(({ key, name }) => {
        return {
          value: key,
          label: name,
        } as ProjectOption;
      }),
    ]);
```

Reply:
Good catch! I was not aware I could simply return the promise. 👍🏼

Reply:
Tested and it is not working because of the debounce. However, I could simplify the usage of `callback` anyways. It looks better and works just like the API describes.